src/main/java/com/autotune/experimentManager/core/EMTransitionRegistry.java

```diff
@@ -16,6 +18,22 @@ public static EMUtil.EMExpStages getNextStage(EMUtil.EMExpStages currentStage) {
         return EMUtil.EMExpStages.get(currentStage.ordinal() + 1);
     }
 
+    public static EMUtil.EMExpStages getNextStage(EMUtil.EMExpStages currentStage, String runId) {
+        if (EMUtil.EMExpStages.getSize() <= (currentStage.ordinal() + 1)){
+            return currentStage;
+        }
+        if (currentStage.isCycle()) {
+            // Implement the breaking logic. Might be dependent on ETD
+            ExperimentTrialData currentETD = ((ExperimentTrialData) EMMapper.getInstance().getMap().get(runId));
+            boolean breakCycle = EMUtil.breakingCondition(currentStage, currentETD);
+            if (!breakCycle) {
+                return currentStage;
+            }
+        }
+        return EMUtil.EMExpStages.get(currentStage.ordinal() + 1);
+    }
+
+
     public static EMUtil.EMExpStages getPossibleContinuousStage(EMUtil.EMExpStages currentStage) {
         EMUtil.EMExpStages previousStage = currentStage;
         for (int i = currentStage.ordinal() + 1; i < EMUtil.EMExpStages.getSize(); i++) {
```

Comment:
Wondering if we are adding 1 as it starts from 0, then should it not be `<=` getSize?

Reply:
Thanks for the review @dinogun 

A little background for the method:

The `getPossibleContinuousStage` is added to handle a future scenerio where if there are two continous independent stages which can be processed parallel ignoring if they have any intermediate `scheduled` stage. This usecase is not occured yet but EM is designed to have a handler for such scenerios.

> if we are adding 1 as it starts from 0

we are adding 1 to check from the next stages of the current stage.

> then should it not be <= getSize?

I would rather leave it with `<` getSize because we might hit an index out of range exception if `i` is equal to getSize.
The next instruction checks if `list[i]` is a scheduled transition

Note: Currently there is no usage in EM for `getPossibleContinuousStage`